ext/pcntl/pcntl.c

```diff
@@ -385,6 +385,55 @@ PHP_FUNCTION(pcntl_waitpid)
 }
 /* }}} */
 
+/* {{{ Waits on or returns the status of a forked child as defined by the waitid() system call */
+PHP_FUNCTION(pcntl_waitid)
+{
+	zend_long idtype = P_PID;
+	zend_long id = 0;
+	/* Optional by-ref array of ints */
+	zval *user_siginfo = NULL;
+	zend_long options = WEXITED;
+	int success;
+
+	ZEND_PARSE_PARAMETERS_START(2, 4)
+		Z_PARAM_LONG(idtype)
+		Z_PARAM_LONG(id)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_ZVAL(user_siginfo)
+		Z_PARAM_LONG(options)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (idtype != P_ALL && idtype != P_PID && idtype != P_PGID && idtype != P_PIDFD) {
```

Comment:
last thing: do not forget what I mentioned about non standard flags. You may develop under a recent Linux distro but not the case for everyone else. macOs users, BSD ... you need to see the multiplatform context. Or if you do not want, you can just deal with only standard flags and someone else will deal with other oses, no big deal :).

Reply:
Agreed, both idtype and flags are platform-specific.

It seems that the best course is to leave validation to the `waitid` call itself and let it set `errno`.